### PR TITLE
created travis build script

### PR DESCRIPTION
**Pull request by fabolhak**

### CODE REVIEW

**Comment by mp4096** [comment]
:+1: I'm also for using something simpler than `sr`'s mega Swiss knife.

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ env:
     - ros_release_name: kinetic
     - ubuntu_version_name: xenial
     - used_modules: check_build #check_cache,code_style_check,check_deb_make,codecov_tool
-    - remote_shell_script: 'https://raw.githubusercontent.com/shadow-robot/sr-build-tools/$toolset_branch/bin/sr-run-ci-build.sh'
+    - remote_shell_script: 'https://raw.githubusercontent.com/tum-phoenix/drive_ros_config/travis-build-script/.travis-build.sh'
```
Comment (mp4096) [inline .travis.yml:15]: You can just use the local script directly.

I'd rather create a dedicated `ci` folder and put the script into it.

File: .travis-build.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+export ubuntu_version=${ubuntu_version_name:-"trusty"}
+export ros_release=${ros_release_name:-"indigo"}
+export docker_image=${docker_image_name:-"shadowrobot/build-tools:$ubuntu_version-$ros_release"}
+
+export docker_user=${docker_user_name:-"user"}
+export docker_user_home=${docker_user_home_dir:-"/home/user"}
+
+export extra_variables="codecov_secure=$CODECOV_TOKEN github_login=$GITHUB_LOGIN github_password=$GITHUB_PASSWORD ros_release=$ros_release ubuntu_version_name=$ubuntu_version "
```
Comment (mp4096) [inline .travis-build.sh:18]: Do we really need all this stuff? Handling private tokens in GitHub/Travis/AppVeyor is a potential minefield. If you need tokens, you should [encrypt them](https://docs.travis-ci.com/user/encryption-keys).

E.g. something could leak through Travis logs when the envvars get replaced within the string.

**Review by mp4096 — COMMENTED** [review]
I guess this is a WIP, but I just leave my comments here.

File: .travis-build.sh (hunk 2 above)
Comment (fabolhak) [reply .travis-build.sh:18]: good point. I don't think we need them. I will check if we can remove them.

Ultimatetly I would like to move all the stuff from the *.sh to the travis.yml file.

Comment (mp4096) [reply .travis-build.sh:18]: I can understand your motivation of putting everything into `.travis.yml`. OTOH, sometimes it's nice to have a 'do everything' script.

If you put everything into `.travis.yml`, you might find YAML's multiline string literals with `>` and `|` handy.

File: .travis.yml (hunk 1 above)
Comment (mp4096) [reply .travis.yml:15]: Uhm, I mean you could just run this script directly instead of `curl`-ing it from GitHub.

File: .travis-build.sh (hunk 2 above)
Comment (fabolhak) [reply .travis-build.sh:18]: yeah. I have to read up on how to format the travis.yml files first.

Comment (mp4096) [reply .travis-build.sh:18]: Some tips:
* https://learnxinyminutes.com/docs/yaml/
* https://docs.travis-ci.com/user/customizing-the-build/#The-Build-Lifecycle

**Pull request merged**